modules/storages/app/controllers/storages/admin/storages_controller.rb

```diff
@@ -86,11 +86,13 @@ def new
   def create
     service_result = Storages::Storages::CreateService.new(user: current_user).call(permitted_storage_params)
     @object = service_result.result
+    @oauth_application = result_from_first_dependent_service(service_result)
 
-    if service_result.success? && (@oauth_application = service_result.dependent_results&.first&.result)
+    if service_result.success? && @oauth_application
       flash[:notice] = I18n.t(:notice_successful_create)
       render :show_oauth_application
     else
+      service_result.merge_dependent_errors!
```

Comment:
See comment above. There is no need to mutate the `ServiceResult` you just want to access the flattened errors. Until further need of mutating or passing this `ServiceResult` on, I would like to have this list of errors computed on demand.